docs/builders/builder-api.md

```diff
@@ -0,0 +1,163 @@
+---
+title: 'Builder API'
+---
+
+Storybook is architected to support multiple builders, including [Webpack](https://webpack.js.org/), [Vite](https://vitejs.dev/), and [ESBuild](https://esbuild.github.io/). The builder API is the set of interfaces to add a new builder to Storybook.
+
+![Storybook builders](./storybook-builders.png)
+
+## How do builders work?
+
+In Storybook, a builder is responsible for compiling your components and stories into JS bundles that run in the browser. A builder also provides a development server for interactive development and a production mode for optimized bundles.
+
+To opt into a builder, the user must add it as a dependency and then edit their configuration file (`.storybook/main.js`) to enable it. For example, with the Vite builder:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-vite-builder-install.yarn.js.mdx',
+    'common/storybook-vite-builder-install.npm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-vite-builder-register.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+## Builder API
+
+In Storybook, every builder must implement the following [API](https://github.com/storybookjs/storybook/blob/next/lib/core-common/src/types.ts#L170-L189), exposing the following configuration options and entry points:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-builder-api-interface.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+In development mode, the `start` API call is responsible for initializing the development server to monitor the file system for changes (for example, components, stories) and execute a hot module reload in the browser.
+It also provides a **bail** function to allow the running process to end gracefully, either via user input or error.
+
+In production, the `build` API call is responsible for generating a static Storybook build, storing it by default in the `storybook-static` directory if no additional configuration is provided. The generated output should contain everything the user needs to view its Storybook by opening either the `index.html` or `iframe.html` in a browser with no other processes running.
```

Comment:
Should we add the whole path here to indicate that it is a directory?

For instance: `.storybook/storybook-static` or `/storybook-static` 
(Not sure where `storybook-static` gets put by default.)

Reply:
Actually no,`storybook-static` by default is put on the root level of the project/app. But there could be the case that they can run the build command passing additional CLI options and placing it in another place, they could have set up a monorepo with NX/Turbosap and have multiple Storybook instances each placing them into separate directories. Hence why I left it like this.